widgetssdk/src/main/java/com/glia/widgets/view/unifiedui/theme/defaulttheme/Survey.kt

```diff
@@ -0,0 +1,67 @@
+@file:Suppress("FunctionName")
+
+package com.glia.widgets.view.unifiedui.theme.defaulttheme
+
+import android.graphics.Color
+import com.glia.widgets.view.unifiedui.exstensions.composeIfAtLeastOneNotNull
+import com.glia.widgets.view.unifiedui.theme.ColorPallet
+import com.glia.widgets.view.unifiedui.theme.base.LayerTheme
+import com.glia.widgets.view.unifiedui.theme.base.TextTheme
+import com.glia.widgets.view.unifiedui.theme.survey.*
+
+/**
+ * Default theme for Survey screen
+ */
+internal fun SurveyDefaultTheme(pallet: ColorPallet): SurveyTheme =
+    SurveyTheme(
+        layer = LayerTheme(fill = pallet.backgroundColorTheme),
+        title = TextTheme(textColor = pallet.baseNormalColorTheme),
+        submitButton = PositiveDefaultButton(pallet),
+        cancelButton = NegativeDefaultButton(pallet),
+        booleanQuestion = SurveyBooleanQuestionTheme(
+            title = TextTheme(textColor = pallet.baseDarkColorTheme),
+            SurveyDefaultOptionButtonTheme(pallet)
+        ),
+        scaleQuestion = SurveyScaleQuestionTheme(
+            title = TextTheme(textColor = pallet.baseDarkColorTheme),
+            SurveyDefaultOptionButtonTheme(pallet)
+        ),
+        singleQuestion = SurveySingleQuestionTheme(
+            title = TextTheme(textColor = pallet.baseDarkColorTheme),
+            tintColor = pallet.primaryColorTheme,
+            option = TextTheme(textColor = pallet.baseDarkColorTheme)
+        ),
+        inputQuestion = SurveyInputQuestionTheme(
+            title = TextTheme(textColor = pallet.baseDarkColorTheme),
+            option = SurveyDefaultOptionButtonTheme(pallet),
+            text = TextTheme(textColor = pallet.baseDarkColorTheme)
+        )
+    )
+
+/**
+ * Default theme for Survey option button
+ */
+internal fun SurveyDefaultOptionButtonTheme(pallet: ColorPallet): OptionButtonTheme? = pallet.run {
+    composeIfAtLeastOneNotNull(
+        baseDarkColorTheme,
+        baseNormalColorTheme,
+        primaryColorTheme,
+        baseLightColorTheme,
+        systemNegativeColorTheme
+    ) {
+        OptionButtonTheme(
+            normalText = TextTheme(textColor = baseDarkColorTheme),
+            normalLayer = LayerTheme(
+                fill = backgroundColorTheme,
+                stroke = baseNormalColorTheme?.primaryColor
+            ),
+            selectedText = TextTheme(textColor = baseLightColorTheme),
+            selectedLayer = LayerTheme(fill = primaryColorTheme, stroke = Color.TRANSPARENT),
+            highlightedText = TextTheme(textColor = systemNegativeColorTheme),
+            highlightedLayer = LayerTheme(
+                fill = backgroundColorTheme,
+                stroke = systemNegativeColorTheme?.primaryColor
+            )
```

Comment:
All of this seems very confusing ))
I guess once all of this is done it would need a lot of manual testing

Reply:
Yes, there is no way to compose default theme easier